mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+        self.filterLayout = 'kcyx'
+        self.inputLayout = 'nchw'
+        self.outputLayout = 'nkhw'
+        self.n = 128
+        self.c = 1024
+        self.hi = 14
+        self.wi = 14
+        self.k = 1024
+        self.y = 1
+        self.x = 1
+        self.convStrideH = 1
+        self.convStrideW = 1
+        self.paddingH = 0
+        self.paddingW = 0
+        self.dilationH = 1
+        self.dilationW = 1
+        self.group = 1
+
+        mlirFilterLayout={"NCHW":"kcyx", "NHWC":"kyxc"}
+        mlirOutputLayout={"NCHW":"nkhw", "NHWC":"nkhw"}
+        # determine dataType from argv[1]
+        if argv[0] == 'conv':
+            self.dataType = 'f32'
+        elif argv[0] == 'convfp16':
+            self.dataType = 'f16'
+        elif argv[0] == 'convbfp16':
+            self.dataType = 'bf16'
+
+        try:
+            # TBD:
+            # implement -m ?
+            # implement -t ?
+            opts, args = getopt.getopt(argv[1:], "F:f:I:O:n:c:H:W:k:y:x:p:q:l:j:u:v:g:m:t:")
+        except getopt.GetOptError:
+            print('getopt error')
+            sys.exit(-1)
+
+        for opt, arg in opts:
+            if opt == '-F':
+                # -F
+                # 1 fwd only
+                # 2 bwd only
+                # 4 wrw only
+                # TBD:
+                # 0 fwd+bwd+wrw
+                # 3 fwd+bwd
+                # 5 fwd+wrw
+                # 6 bwd+wrw
+                if int(arg) == 1:
+                    self.direction = 'fwd'
+                elif int(arg) == 2:
+                    self.direction = 'bwd'
+                elif int(arg) == 4:
+                    self.direction = 'wrw'
+            elif opt == '-f':
+                # -f
+                self.filterLayout = mlirFilterLayout[arg]
+            elif opt == '-I':
+                # -I
+                self.inputLayout = arg.lower()
+            elif opt == '-O':
+                # -O
+                self.outputLayout = mlirOutputLayout[arg]
+            elif opt == "-n":
+                # -n
+                self.n = int(arg)
+            elif opt == '-c':
+                # -c
+                self.c = int(arg)
+            elif opt == '-H':
+                # -H
+                self.hi = int(arg)
+            elif opt == '-W':
+                # -W
+                self.wi = int(arg)
+            elif opt == '-k':
+                # -k
+                self.k = int(arg)
+            elif opt == '-y':
+                # -y
+                self.y = int(arg)
+            elif opt == '-x':
+                # -x
+                self.x = int(arg)
+            elif opt == '-u':
+                # -u
+                self.convStrideH = int(arg)
+            elif opt == '-v':
+                # -v
+                self.convStrideW = int(arg)
+            elif opt == '-p':
+                # -p
+                self.paddingH = int(arg)
+            elif opt == '-q':
+                # -q
```

Comment:
These comments don't add anything

Reply:
That is, we already know what `opt` is from the if statement. Could you remove them or, alternatively, expand on what each option does?